CHANGELOG.md

```diff
@@ -48,6 +48,12 @@ _None._
 
 _None._
 
+## [2.0.0-beta.2](https://github.com/wordpress-mobile/WordPress-iOS-Shared/releases/tag/2.0.0-beta.2)
```

Comment:
We are not supposed to add the version in the title of the header until we do an actual release, because if the PR were to land later than anticipated and after another PR that happened to add, say, a breaking change, or if we happen not to do a release immediately after this PR lands but waited for some others to be added before doing a release, that forecasted version you put in the header would be affected and possibly not be the version number you initially thought it'd be.

So since we can only know what version number to use (in order to comply to SemVer) at the time we actually do the releases and see the whole set of changes done since last release, the practice is supposed to be to keep the title as `Unreleased` when adding items to the CHANGELOG during individual PRs, and only set the title in the PR doing the release.

---

All that being said, I'm not sure if we have an official stance for betas. Maybe @mokagio  can come in and give his insight? My personal take would be to keep `Unreleased` title until we do a final version, and thus have in the CHANGELOG only section titles that are final versions, not `*beta*` (in other words, not consider a beta tag to be a release in the sense of the CHANGELOG). And even eventually consider ditching the practice to do `*beta*` tags (because they are also subject to the problem from above about not doing the right SemVer bump (because `SemVerBump(Change1) + SemVerBump(Change2) ≠ SemVerBump(Change1 + Change2)`) and instead have clients point to a commitish (sha1) during development, and only do (non-beta) lib releases (and update the `Unreleased` title) during code freeze. Just like we've been doing for a long while in the Android side. At least I thought that was the plan, just not sure if it has been officialized yet.

Reply:
Lol I should have refreshed the page before submitting my prose, I see @mokagio ended up asking a similar (but more concise 😅) question [here](https://github.com/wordpress-mobile/WordPress-iOS-Shared/pull/327#discussion_r1034624508)… great minds 😄

Reply:
> My personal take would be to keep Unreleased title until we do a final version

I'd prefer this as well.

Reply:
> We are not supposed to add the version in the title of the header until we do an actual release

I should've mentioned it in the PR description. I took a shortcut here. Instead of waiting for this PR to be merged and creating a separate PR to bump the version and update changelog (like I did for [2.0.0-beta.1](https://github.com/wordpress-mobile/WordPress-iOS-Shared/pull/326)), I did them all in one PR. So that I can release a new beta and unblock https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/pull/708 and https://github.com/wordpress-mobile/WordPressKit-iOS/pull/550 straight after this PR is merged.

I agree with your point about a potentially conflict with other PRs. But in this particular instance, this repo doesn't get PRs very often, which is the major reason I thought it's okay to include the "release changes" in this PR.

In regards to including betas in CHANGELOG.md or not, my thinking is versions in CHANGELOG.md should match the ones in [the GitHub Releases page](https://github.com/wordpress-mobile/WordPress-iOS-Shared/releases). If it's a standard practice to do beta releases, then we need to include betas in the CHANGELOG.md. But I don't have a strong opinion on whether we should do beta or not 😄